### PR TITLE
Export *STM functions

### LINKED ISSUE #5: Expose *STM functions
As an API consumer building on top of this library it would be useful to have access to the underlying STM functions. For instance, an implementation for inserting a key if there is 'enough space' might look like follows:
```
insertIfSpace :: Int -> k -> v -> Cache k v -> Maybe TimeSpec -> STM ()
insertIfSpace maxSize k v cache mexpiry = do
  t <- now
  atomically $ do
    -- purge expired keys to free up space
    ks <- keysSTM cache
    forM_ ks $ \k -> lookupItemT True k cache t

    -- insert if doing so would not make the cache too big
    sz <- sizeSTM cache
    when (sz < maxSize) $ do
      insertT k v cache mexpiry
```
Unfortunately only having access to the IO functions means that this can't be implemented in a transactionally safe manner.

### PR DESCRIPTION
**Pull request by hverr**
Closes #5 

cc: @charles-cooper: could you please confirm this suits your needs?

### CODE REVIEW

**Comment by charles-cooper** [comment]
Hey @hverr, very cool! I think a couple things might additionally help - exposing the `Cache` constructor (maybe in a `Data.Cache.Internal` module) for low level access and also a `nowSTM = unsafeIOToSTM` since `now` doesn't have any (memory) side effects.

**Comment by hverr** [comment]
@charles-cooper Please check 7d8abc821ec58e185bcc42ea5b50e67ab09e88a0

File: src/Data/Cache/Internal.hs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+-- |
+-- Module:      Data.Cache.Internal
+-- Copyright:   (c) 2017 Henri Verroken
+-- LIcense:     BSD3
```
Comment (charles-cooper) [inline src/Data/Cache/Internal.hs]: License

**Comment by hverr** [comment]
Cool! Thanks! Waiting for travis build, will then bump version and upload to hackage.

**Comment by charles-cooper** [comment]
Thanks @hverr!! It looks great.

**Pull request merged**